### PR TITLE
Breaking change but a good change:  Cleared up the API  for adding sinks

### PR DESCRIPTION
**Pull request by KjellKod**
Breaking change but a good change:  Cleared up the API for Creating log worker and Adding sinks

### CODE REVIEW

File: src/g3log/logworker.hpp [hunk 1]
```diff
@@ -46,33 +41,60 @@ namespace g3 {
...
 
+      /// Creates the LogWorker with no sinks. See exampel below on @ref addSink for how to use it
+      /// if you want to use the default file logger then see below for @ref addDefaultLogger
+      static std::unique_ptr<LogWorker> createLogWorker();
+
+      
+      /**
+      A convenience function to add the default g3::FileSink to the log worker
+       
+       @param the worker (no, don't put in nullptr here!)
```
Comment (KjellKod) [inline src/g3log/logworker.hpp]: remove line

File: example/main_fatal_choice.cpp [hunk 2]
```diff
@@ -242,11 +246,11 @@ void breakHere() {
 
 int main(int argc, char **argv)
 {
-   auto logger_n_handle = g3::LogWorker::createWithDefaultLogger(argv[0], path_to_log_file);
-   g3::initializeLogging(logger_n_handle.worker.get());
-   g3::setFatalPreLoggingHook(&breakHere);
```
Comment (KjellKod) [inline example/main_fatal_choice.cpp]: put this back

File: src/g3log/logworker.hpp [hunk 3]
```diff
@@ -46,33 +41,60 @@ namespace g3 {
...
+       @verbatim
+       Example:
+       using namespace g3;
+       std::unique_ptr<LogWorker> logworker {LogWorker::createLogWorker()};
+       auto handle = addDefaultLogger("my_test_log", "/tmp");
+       initializeLogging(logworker.get()); // ref. g3log.hpp
+
+       std::future<std::string> log_file_name = sinkHandle->call(&FileSink::fileName);
+       std::cout << "The filename is: " << log_file_name.get() << std::endl;
+       //   something like: /tmp/
```
Comment (KjellKod) [inline src/g3log/logworker.hpp]: add example of file name

File: example/main_fatal_choice.cpp (hunk 2 above)
Comment (KjellKod) [inline example/main_fatal_choice.cpp]: done: Line 253

**Comment by KjellKod** [comment]
:snail:  --> :rabbit2: 
 pending feedback from @will2142 , @AlphaCluster

**Comment by AlphaCluster** [comment]
I like it I think it simplifies the initialization of the log worker.

**Pull request merged**